NuRadioReco/modules/base/module.py

```diff
@@ -0,0 +1,76 @@
+from functools import wraps
+from timeit import default_timer as timer
+
+def run_decorator(run):
+    @wraps(run)
+    def register_run_method(self, evt, station, det, **kwargs):
+        evt.register_module(self, self.__class__.__name__, kwargs)
+        start = timer()
+        res =  run(self, evt, station, det, **kwargs)
+        end = timer()
+        if not self in register_run_method.time:
+            register_run_method.time[self] = 0
+        register_run_method.time[self] += (end - start)
+        return res
+    register_run_method.time = {}
+    return register_run_method
+       
+
+
+def time_this(original_function):      
+    print("decorating")
+    def new_function(*args,**kwargs):
+        print("starting timer")
+        import datetime                 
+        before = datetime.datetime.now()                     
+        x = original_function(*args,**kwargs)                
+        after = datetime.datetime.now()                      
+        print("Elapsed Time = {0}".format(after-before))
+        return x                                             
+    return new_function  
+
+def ModuleDecorator(Cls):
```

Comment:
Do you actually need this? So far you have only used the run_decorator. If you want to use it, it should adhere a bit more to our coding standards. No print statements, NewCls should be changed in naming, etc.

Reply:
No it is not used, although it would be more elegant to make a class decorator for the full module, but I didn't get it to work...